### ISSUE TITLE
Improve Unit Tests - NPE in Some Cases

### ISSUE THREAD

**Issue by jeremylong** [dependency-check/DependencyCheck#181]
A generic utility method needs to be added to the BaseTest classes to obtain references to jar files used when testing that uses the [junit assume](http://junit.sourceforge.net/javadoc/org/junit/Assume.html) methods to ensure that the files exist. This new method should be used in the test cases anywhere code obtains the location of a jar file such as:

``` java
File struts = new File(this.getClass().getClassLoader().getResource("struts2-core-2.1.2.jar").getPath());
```

Additionally, consider adding a PMD rule to flag test cases that directly call getClassLoader().getResource().

**Comment by hansjoachim** [comment]
I agree that this can be split out in a separate method. However, I am not sure about using assume, since it would mean the tests are marked as ignored if the files are not found. 

At least when I ran into problems with this (HintAnalyzerTest), I got it consistently when using openjdk-8 (tried Ubuntu and Arch Linux, both result in null pointer in this test case). But with openjdk-7 all tests passed. So I wonder if there might be some difference between jdk7 and 8 which might cause this.
